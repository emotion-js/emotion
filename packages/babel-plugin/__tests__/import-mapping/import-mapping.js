import babelTester from 'babel-tester'
import plugin from '@emotion/babel-plugin'

babelTester('import mapping', __dirname, {
  plugins: [
    [
      plugin,
      {
        importMap: {
          'package-one': {
            nonDefaultStyled: {
              canonicalImport: ['@emotion/styled', 'default']
            }
          },
          'package-two': {
            someJsx: {
              canonicalImport: ['@emotion/react', 'jsx']
            },
            someCssFromCore: {
              canonicalImport: ['@emotion/react', 'css']
            },
            someKeyframesFromCore: {
              canonicalImport: ['@emotion/react', 'keyframes']
            },
            SomeGlobalFromCore: {
              canonicalImport: ['@emotion/react', 'Global']
            }
          },

          'package-three': {
            something: {
              canonicalImport: ['@emotion/css', 'css']
            },
            keyframes: {
              canonicalImport: ['@emotion/css', 'keyframes']
            }
          },
          'package-four': {
            nonDefaultStyled: {
              canonicalImport: ['@emotion/styled', 'default'],
              styledBaseImport: ['package-four/base', 'something']
            }
          }
        }
      }
    ]
  ]
})
