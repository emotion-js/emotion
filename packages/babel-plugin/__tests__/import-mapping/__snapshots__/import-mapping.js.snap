// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import mapping emotion-css-keyframes 1`] = `
"import { keyframes } from 'package-three'

keyframes\`
  from, 20%, 53%, 80%, to {
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    transform: translate3d(0, -30px, 0);
  }

  70% {
    transform: translate3d(0, -15px, 0);
  }

  90% {
    transform: translate3d(0,-4px,0);
  }
\`


      ↓ ↓ ↓ ↓ ↓ ↓

function _EMOTION_STRINGIFIED_CSS_ERROR__() { return \\"You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).\\"; }

import { keyframes } from 'package-three';

/*#__PURE__*/
keyframes(process.env.NODE_ENV === \\"production\\" ? {
  name: \\"w3sfyc\\",
  styles: \\"from,20%,53%,80%,to{transform:translate3d(0,0,0);}40%,43%{transform:translate3d(0, -30px, 0);}70%{transform:translate3d(0, -15px, 0);}90%{transform:translate3d(0,-4px,0);}\\"
} : {
  name: \\"w3sfyc\\",
  styles: \\"from,20%,53%,80%,to{transform:translate3d(0,0,0);}40%,43%{transform:translate3d(0, -30px, 0);}70%{transform:translate3d(0, -15px, 0);}90%{transform:translate3d(0,-4px,0);}\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImVtb3Rpb24tY3NzLWtleWZyYW1lcy5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFFUyIsImZpbGUiOiJlbW90aW9uLWNzcy1rZXlmcmFtZXMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBrZXlmcmFtZXMgfSBmcm9tICdwYWNrYWdlLXRocmVlJ1xuXG5rZXlmcmFtZXNgXG4gIGZyb20sIDIwJSwgNTMlLCA4MCUsIHRvIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsMCwwKTtcbiAgfVxuXG4gIDQwJSwgNDMlIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsIC0zMHB4LCAwKTtcbiAgfVxuXG4gIDcwJSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLCAtMTVweCwgMCk7XG4gIH1cblxuICA5MCUge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwtNHB4LDApO1xuICB9XG5gXG4iXX0= */\\",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});"
`;

exports[`import mapping emotion-react-keyframes 1`] = `
"import * as React from 'react'
import { someCssFromCore, someKeyframesFromCore } from 'package-two'

const numForty = 40
const numThirty = 30

const bounce = someKeyframesFromCore\`
  from, 20%, 53%, 80%, to {
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    transform: translate3d(0, -30px, 0);
  }

  \${numForty + numThirty}% {
    transform: translate3d(0, -15px, 0);
  }

  90% {
    transform: translate3d(0,-4px,0);
  }
\`

const gift = someKeyframesFromCore\`
  from {
    transform: translate3d(0,0,0);
  }

  to {
    transform: translate3d(0,-4px,0);
  }
\`

const Comp = () => (
  <div
    className={someCssFromCore\`
      animation: \${bounce} 1s ease infinite;
    \`}
  >
    ichenlei with emotion
  </div>
)

const Comp2 = () => (
  <div
    className={someCssFromCore\`
      animation: \${gift} 1s ease infinite;
    \`}
  >
    ichenlei with emotion
  </div>
)


      ↓ ↓ ↓ ↓ ↓ ↓

function _EMOTION_STRINGIFIED_CSS_ERROR__() { return \\"You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).\\"; }

import * as React from 'react';
import { someCssFromCore, someKeyframesFromCore } from 'package-two';
const numForty = 40;
const numThirty = 30;
const bounce = /*#__PURE__*/someKeyframesFromCore(\\"from,20%,53%,80%,to{transform:translate3d(0,0,0);}40%,43%{transform:translate3d(0, -30px, 0);}\\", numForty + numThirty, \\"%{transform:translate3d(0, -15px, 0);}90%{transform:translate3d(0,-4px,0);}\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImVtb3Rpb24tcmVhY3Qta2V5ZnJhbWVzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQU1vQyIsImZpbGUiOiJlbW90aW9uLXJlYWN0LWtleWZyYW1lcy5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAqIGFzIFJlYWN0IGZyb20gJ3JlYWN0J1xuaW1wb3J0IHsgc29tZUNzc0Zyb21Db3JlLCBzb21lS2V5ZnJhbWVzRnJvbUNvcmUgfSBmcm9tICdwYWNrYWdlLXR3bydcblxuY29uc3QgbnVtRm9ydHkgPSA0MFxuY29uc3QgbnVtVGhpcnR5ID0gMzBcblxuY29uc3QgYm91bmNlID0gc29tZUtleWZyYW1lc0Zyb21Db3JlYFxuICBmcm9tLCAyMCUsIDUzJSwgODAlLCB0byB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLDAsMCk7XG4gIH1cblxuICA0MCUsIDQzJSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLCAtMzBweCwgMCk7XG4gIH1cblxuICAke251bUZvcnR5ICsgbnVtVGhpcnR5fSUge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwgLTE1cHgsIDApO1xuICB9XG5cbiAgOTAlIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsLTRweCwwKTtcbiAgfVxuYFxuXG5jb25zdCBnaWZ0ID0gc29tZUtleWZyYW1lc0Zyb21Db3JlYFxuICBmcm9tIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsMCwwKTtcbiAgfVxuXG4gIHRvIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsLTRweCwwKTtcbiAgfVxuYFxuXG5jb25zdCBDb21wID0gKCkgPT4gKFxuICA8ZGl2XG4gICAgY2xhc3NOYW1lPXtzb21lQ3NzRnJvbUNvcmVgXG4gICAgICBhbmltYXRpb246ICR7Ym91bmNlfSAxcyBlYXNlIGluZmluaXRlO1xuICAgIGB9XG4gID5cbiAgICBpY2hlbmxlaSB3aXRoIGVtb3Rpb25cbiAgPC9kaXY+XG4pXG5cbmNvbnN0IENvbXAyID0gKCkgPT4gKFxuICA8ZGl2XG4gICAgY2xhc3NOYW1lPXtzb21lQ3NzRnJvbUNvcmVgXG4gICAgICBhbmltYXRpb246ICR7Z2lmdH0gMXMgZWFzZSBpbmZpbml0ZTtcbiAgICBgfVxuICA+XG4gICAgaWNoZW5sZWkgd2l0aCBlbW90aW9uXG4gIDwvZGl2PlxuKVxuIl19 */\\"));
const gift = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"15er54a\\",
  styles: \\"@keyframes 15er54a{from{transform:translate3d(0,0,0);}to{transform:translate3d(0,-4px,0);}}\\",
  anim: 1
} : {
  name: \\"15er54a\\",
  styles: \\"@keyframes 15er54a{from{transform:translate3d(0,0,0);}to{transform:translate3d(0,-4px,0);}}\\",
  anim: 1,
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImVtb3Rpb24tcmVhY3Qta2V5ZnJhbWVzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQXdCa0MiLCJmaWxlIjoiZW1vdGlvbi1yZWFjdC1rZXlmcmFtZXMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgKiBhcyBSZWFjdCBmcm9tICdyZWFjdCdcbmltcG9ydCB7IHNvbWVDc3NGcm9tQ29yZSwgc29tZUtleWZyYW1lc0Zyb21Db3JlIH0gZnJvbSAncGFja2FnZS10d28nXG5cbmNvbnN0IG51bUZvcnR5ID0gNDBcbmNvbnN0IG51bVRoaXJ0eSA9IDMwXG5cbmNvbnN0IGJvdW5jZSA9IHNvbWVLZXlmcmFtZXNGcm9tQ29yZWBcbiAgZnJvbSwgMjAlLCA1MyUsIDgwJSwgdG8ge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwwLDApO1xuICB9XG5cbiAgNDAlLCA0MyUge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwgLTMwcHgsIDApO1xuICB9XG5cbiAgJHtudW1Gb3J0eSArIG51bVRoaXJ0eX0lIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsIC0xNXB4LCAwKTtcbiAgfVxuXG4gIDkwJSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLC00cHgsMCk7XG4gIH1cbmBcblxuY29uc3QgZ2lmdCA9IHNvbWVLZXlmcmFtZXNGcm9tQ29yZWBcbiAgZnJvbSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLDAsMCk7XG4gIH1cblxuICB0byB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLC00cHgsMCk7XG4gIH1cbmBcblxuY29uc3QgQ29tcCA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNsYXNzTmFtZT17c29tZUNzc0Zyb21Db3JlYFxuICAgICAgYW5pbWF0aW9uOiAke2JvdW5jZX0gMXMgZWFzZSBpbmZpbml0ZTtcbiAgICBgfVxuICA+XG4gICAgaWNoZW5sZWkgd2l0aCBlbW90aW9uXG4gIDwvZGl2PlxuKVxuXG5jb25zdCBDb21wMiA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNsYXNzTmFtZT17c29tZUNzc0Zyb21Db3JlYFxuICAgICAgYW5pbWF0aW9uOiAke2dpZnR9IDFzIGVhc2UgaW5maW5pdGU7XG4gICAgYH1cbiAgPlxuICAgIGljaGVubGVpIHdpdGggZW1vdGlvblxuICA8L2Rpdj5cbilcbiJdfQ== */\\",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

const Comp = () => <div className={/*#__PURE__*/someCssFromCore(\\"animation:\\", bounce, \\" 1s ease infinite;\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\";label:Comp;\\"), process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImVtb3Rpb24tcmVhY3Qta2V5ZnJhbWVzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQW9DOEIiLCJmaWxlIjoiZW1vdGlvbi1yZWFjdC1rZXlmcmFtZXMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgKiBhcyBSZWFjdCBmcm9tICdyZWFjdCdcbmltcG9ydCB7IHNvbWVDc3NGcm9tQ29yZSwgc29tZUtleWZyYW1lc0Zyb21Db3JlIH0gZnJvbSAncGFja2FnZS10d28nXG5cbmNvbnN0IG51bUZvcnR5ID0gNDBcbmNvbnN0IG51bVRoaXJ0eSA9IDMwXG5cbmNvbnN0IGJvdW5jZSA9IHNvbWVLZXlmcmFtZXNGcm9tQ29yZWBcbiAgZnJvbSwgMjAlLCA1MyUsIDgwJSwgdG8ge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwwLDApO1xuICB9XG5cbiAgNDAlLCA0MyUge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwgLTMwcHgsIDApO1xuICB9XG5cbiAgJHtudW1Gb3J0eSArIG51bVRoaXJ0eX0lIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsIC0xNXB4LCAwKTtcbiAgfVxuXG4gIDkwJSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLC00cHgsMCk7XG4gIH1cbmBcblxuY29uc3QgZ2lmdCA9IHNvbWVLZXlmcmFtZXNGcm9tQ29yZWBcbiAgZnJvbSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLDAsMCk7XG4gIH1cblxuICB0byB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLC00cHgsMCk7XG4gIH1cbmBcblxuY29uc3QgQ29tcCA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNsYXNzTmFtZT17c29tZUNzc0Zyb21Db3JlYFxuICAgICAgYW5pbWF0aW9uOiAke2JvdW5jZX0gMXMgZWFzZSBpbmZpbml0ZTtcbiAgICBgfVxuICA+XG4gICAgaWNoZW5sZWkgd2l0aCBlbW90aW9uXG4gIDwvZGl2PlxuKVxuXG5jb25zdCBDb21wMiA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNsYXNzTmFtZT17c29tZUNzc0Zyb21Db3JlYFxuICAgICAgYW5pbWF0aW9uOiAke2dpZnR9IDFzIGVhc2UgaW5maW5pdGU7XG4gICAgYH1cbiAgPlxuICAgIGljaGVubGVpIHdpdGggZW1vdGlvblxuICA8L2Rpdj5cbilcbiJdfQ== */\\")}>
    ichenlei with emotion
  </div>;

const Comp2 = () => <div className={/*#__PURE__*/someCssFromCore(\\"animation:\\", gift, \\" 1s ease infinite;\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\";label:Comp2;\\"), process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImVtb3Rpb24tcmVhY3Qta2V5ZnJhbWVzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQThDOEIiLCJmaWxlIjoiZW1vdGlvbi1yZWFjdC1rZXlmcmFtZXMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgKiBhcyBSZWFjdCBmcm9tICdyZWFjdCdcbmltcG9ydCB7IHNvbWVDc3NGcm9tQ29yZSwgc29tZUtleWZyYW1lc0Zyb21Db3JlIH0gZnJvbSAncGFja2FnZS10d28nXG5cbmNvbnN0IG51bUZvcnR5ID0gNDBcbmNvbnN0IG51bVRoaXJ0eSA9IDMwXG5cbmNvbnN0IGJvdW5jZSA9IHNvbWVLZXlmcmFtZXNGcm9tQ29yZWBcbiAgZnJvbSwgMjAlLCA1MyUsIDgwJSwgdG8ge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwwLDApO1xuICB9XG5cbiAgNDAlLCA0MyUge1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlM2QoMCwgLTMwcHgsIDApO1xuICB9XG5cbiAgJHtudW1Gb3J0eSArIG51bVRoaXJ0eX0lIHtcbiAgICB0cmFuc2Zvcm06IHRyYW5zbGF0ZTNkKDAsIC0xNXB4LCAwKTtcbiAgfVxuXG4gIDkwJSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLC00cHgsMCk7XG4gIH1cbmBcblxuY29uc3QgZ2lmdCA9IHNvbWVLZXlmcmFtZXNGcm9tQ29yZWBcbiAgZnJvbSB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLDAsMCk7XG4gIH1cblxuICB0byB7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGUzZCgwLC00cHgsMCk7XG4gIH1cbmBcblxuY29uc3QgQ29tcCA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNsYXNzTmFtZT17c29tZUNzc0Zyb21Db3JlYFxuICAgICAgYW5pbWF0aW9uOiAke2JvdW5jZX0gMXMgZWFzZSBpbmZpbml0ZTtcbiAgICBgfVxuICA+XG4gICAgaWNoZW5sZWkgd2l0aCBlbW90aW9uXG4gIDwvZGl2PlxuKVxuXG5jb25zdCBDb21wMiA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNsYXNzTmFtZT17c29tZUNzc0Zyb21Db3JlYFxuICAgICAgYW5pbWF0aW9uOiAke2dpZnR9IDFzIGVhc2UgaW5maW5pdGU7XG4gICAgYH1cbiAgPlxuICAgIGljaGVubGVpIHdpdGggZW1vdGlvblxuICA8L2Rpdj5cbilcbiJdfQ== */\\")}>
    ichenlei with emotion
  </div>;"
`;

exports[`import mapping global 1`] = `
"import * as React from 'react'
import { SomeGlobalFromCore } from 'package-two'

export default () => <SomeGlobalFromCore styles={{ color: 'hotpink' }} />


      ↓ ↓ ↓ ↓ ↓ ↓

function _EMOTION_STRINGIFIED_CSS_ERROR__() { return \\"You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).\\"; }

import * as React from 'react';
import { SomeGlobalFromCore } from 'package-two';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
} : {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImdsb2JhbC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFHeUMiLCJmaWxlIjoiZ2xvYmFsLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICogYXMgUmVhY3QgZnJvbSAncmVhY3QnXG5pbXBvcnQgeyBTb21lR2xvYmFsRnJvbUNvcmUgfSBmcm9tICdwYWNrYWdlLXR3bydcblxuZXhwb3J0IGRlZmF1bHQgKCkgPT4gPFNvbWVHbG9iYWxGcm9tQ29yZSBzdHlsZXM9e3sgY29sb3I6ICdob3RwaW5rJyB9fSAvPlxuIl19 */\\",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

export default (() => <SomeGlobalFromCore styles={_ref} />);"
`;

exports[`import mapping global-needs-css 1`] = `
"import * as React from 'react'
import { SomeGlobalFromCore } from 'package-two'

const getBgColor = () => ({ backgroundColor: '#fff' })

export default () => (
  <SomeGlobalFromCore styles={{ color: 'hotpink', ...getBgColor() }} />
)


      ↓ ↓ ↓ ↓ ↓ ↓

import { someCssFromCore as _css } from \\"package-two\\";
import * as React from 'react';
import { SomeGlobalFromCore } from 'package-two';

const getBgColor = () => ({
  backgroundColor: '#fff'
});

export default (() => <SomeGlobalFromCore styles={/*#__PURE__*/_css({
  color: 'hotpink',
  ...getBgColor()
}, process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImdsb2JhbC1uZWVkcy1jc3MuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBTXNCIiwiZmlsZSI6Imdsb2JhbC1uZWVkcy1jc3MuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgKiBhcyBSZWFjdCBmcm9tICdyZWFjdCdcbmltcG9ydCB7IFNvbWVHbG9iYWxGcm9tQ29yZSB9IGZyb20gJ3BhY2thZ2UtdHdvJ1xuXG5jb25zdCBnZXRCZ0NvbG9yID0gKCkgPT4gKHsgYmFja2dyb3VuZENvbG9yOiAnI2ZmZicgfSlcblxuZXhwb3J0IGRlZmF1bHQgKCkgPT4gKFxuICA8U29tZUdsb2JhbEZyb21Db3JlIHN0eWxlcz17eyBjb2xvcjogJ2hvdHBpbmsnLCAuLi5nZXRCZ0NvbG9yKCkgfX0gLz5cbilcbiJdfQ== */\\")} />);"
`;

exports[`import mapping jsx 1`] = `
"/** @jsx someJsx */
import { someJsx } from 'package-two'

const SomeComponent = props => <div css={{ color: window }} {...props} />


      ↓ ↓ ↓ ↓ ↓ ↓

import { someCssFromCore as _css } from \\"package-two\\";

/** @jsx someJsx */
import { someJsx } from 'package-two';

const SomeComponent = props => <div css={/*#__PURE__*/_css({
  color: window
}, process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\";label:SomeComponent;\\", process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImpzeC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFHb0MiLCJmaWxlIjoianN4LmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3ggc29tZUpzeCAqL1xuaW1wb3J0IHsgc29tZUpzeCB9IGZyb20gJ3BhY2thZ2UtdHdvJ1xuXG5jb25zdCBTb21lQ29tcG9uZW50ID0gcHJvcHMgPT4gPGRpdiBjc3M9e3sgY29sb3I6IHdpbmRvdyB9fSB7Li4ucHJvcHN9IC8+XG4iXX0= */\\")} {...props} />;"
`;

exports[`import mapping non-default-styled 1`] = `
"import { nonDefaultStyled } from 'package-one'

let SomeComp = nonDefaultStyled.div({ color: 'hotpink' })


      ↓ ↓ ↓ ↓ ↓ ↓

function _EMOTION_STRINGIFIED_CSS_ERROR__() { return \\"You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).\\"; }

import { nonDefaultStyled } from 'package-one';
let SomeComp = nonDefaultStyled(\\"div\\", process.env.NODE_ENV === \\"production\\" ? {
  target: \\"e1dq5q4a0\\"
} : {
  target: \\"e1dq5q4a0\\",
  label: \\"SomeComp\\"
})(process.env.NODE_ENV === \\"production\\" ? {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
} : {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIm5vbi1kZWZhdWx0LXN0eWxlZC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFFZSIsImZpbGUiOiJub24tZGVmYXVsdC1zdHlsZWQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBub25EZWZhdWx0U3R5bGVkIH0gZnJvbSAncGFja2FnZS1vbmUnXG5cbmxldCBTb21lQ29tcCA9IG5vbkRlZmF1bHRTdHlsZWQuZGl2KHsgY29sb3I6ICdob3RwaW5rJyB9KVxuIl19 */\\",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});"
`;

exports[`import mapping styled-with-base-specified 1`] = `
"import { nonDefaultStyled } from 'package-four'

let SomeComp = nonDefaultStyled.div({ color: 'hotpink' })


      ↓ ↓ ↓ ↓ ↓ ↓

import { something as _styled } from \\"package-four/base\\";

function _EMOTION_STRINGIFIED_CSS_ERROR__() { return \\"You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).\\"; }

let SomeComp = _styled(\\"div\\", process.env.NODE_ENV === \\"production\\" ? {
  target: \\"e16az05m0\\"
} : {
  target: \\"e16az05m0\\",
  label: \\"SomeComp\\"
})(process.env.NODE_ENV === \\"production\\" ? {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
} : {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlZC13aXRoLWJhc2Utc3BlY2lmaWVkLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUVlIiwiZmlsZSI6InN0eWxlZC13aXRoLWJhc2Utc3BlY2lmaWVkLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgbm9uRGVmYXVsdFN0eWxlZCB9IGZyb20gJ3BhY2thZ2UtZm91cidcblxubGV0IFNvbWVDb21wID0gbm9uRGVmYXVsdFN0eWxlZC5kaXYoeyBjb2xvcjogJ2hvdHBpbmsnIH0pXG4iXX0= */\\",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});"
`;

exports[`import mapping vanilla 1`] = `
"import { something } from 'package-three'

something({ color: 'green' })


      ↓ ↓ ↓ ↓ ↓ ↓

function _EMOTION_STRINGIFIED_CSS_ERROR__() { return \\"You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).\\"; }

import { something } from 'package-three';

/*#__PURE__*/
something(process.env.NODE_ENV === \\"production\\" ? {
  name: \\"v98kxt\\",
  styles: \\"color:green\\"
} : {
  name: \\"v98kxt\\",
  styles: \\"color:green\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInZhbmlsbGEuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBRUEiLCJmaWxlIjoidmFuaWxsYS5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IHNvbWV0aGluZyB9IGZyb20gJ3BhY2thZ2UtdGhyZWUnXG5cbnNvbWV0aGluZyh7IGNvbG9yOiAnZ3JlZW4nIH0pXG4iXX0= */\\",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});"
`;
